api/src/DataPersister/CampCollaborationDataPersister.php

```diff
@@ -60,12 +63,20 @@ public function beforeCreate($data): BaseEntity {
     }
 
     public function afterCreate($data): void {
+        /** @var CampCollaboration $data */
         /** @var User $user */
         $user = $this->security->getUser();
         $emailToInvite = $data->user?->getEmail() ?? $data->inviteEmail;
         if (CampCollaboration::STATUS_INVITED == $data->status && $emailToInvite) {
             $this->mailService->sendInviteToCampMail($user, $data->camp, $data->inviteKey, $emailToInvite);
         }
+
+        $materialList = new MaterialList();
+        $materialList->campCollaboration = $data;
+        $data->camp->addMaterialList($materialList);
+        $this->em->persist($materialList);
+
+        $this->em->flush();
```

Comment:
Das flush braucht es, weil wir hier im afterCreate sind?

Würde es Sinn machen, solche Logik im beforeCreate zu haben, und im afterCreate nur Nebeneffekte wie z.B. E-Mails versenden?

Reply:
In https://github.com/ecamp/ecamp3/issues/2574